lib/spack/spack/schema/packages.py

```diff
@@ -54,6 +54,24 @@
     ]
 }
 
+prefer_and_conflict = {
+    "type": "array",
+    "items": {
+        "oneOf": [
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {
+                    "spec": {"type": "string"},
+                    "message": {"type": "string"},
+                    "when": {"type": "string"},
+                },
+            },
+            {"type": "string"},
+        ]
+    },
+}
```

Comment:
Is there a reason not to allow a single string here?

```yaml
require: "+foo"
```

Reply:
Yes. Talking with users, it seems single strings are a very prone to difficult to debug errors - since they don't compose among different configuration scopes. The usual complaint is having some system requirement expressed as a string, that cannot be extended at user scope and needs then to be replicated (usually that is also reported with a lot of time spent understanding that the system requirement was overridden).

Using list of strings doesn't add that much boilerplate, since it is:
```yaml
require:
- XXX
```
instead of:
```yaml
require: XXX
```
and has the benefit of not creating such pitfalls.